pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -1224,7 +1226,23 @@ public ShutdownService getShutdownService() {
      * @return Hostname or IP address the service advertises to the outside world.
      */
     public static String advertisedAddress(ServiceConfiguration config) {
-        return ServiceConfigurationUtils.getDefaultOrConfiguredAddress(config.getAdvertisedAddress());
```

Comment:
Could you please update the `ServiceConfigurationUtils.getDefaultOrConfiguredAddress` to support get default IP of the listening interface if the advertisedAddress absent but the listening interface present?

And shall we need to avoid users to configure the both listeningInterface and `bindAddress`? Seems setting both at the same time will cause some confusion.